##### Copyright 2018 The TensorFlow Authors.

Licensed under the Apache License, Version 2.0 (the "License");

In [1]:
#@title Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# 텐서플로 2.0 시작하기: 초보자용

<table class="tfo-notebook-buttons" align="left">
  <td>
    <a target="_blank" href="https://www.tensorflow.org/tutorials/quickstart/beginner"><img src="https://www.tensorflow.org/images/tf_logo_32px.png" />TensorFlow.org에서 보기</a>
  </td>
  <td>
    <a target="_blank" href="https://colab.research.google.com/github/tensorflow/docs-l10n/blob/master/site/ko/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />구글 코랩(Colab)에서 실행하기</a>
  </td>
  <td>
    <a target="_blank" href="https://github.com/tensorflow/docs-l10n/blob/master/site/ko/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />깃허브(GitHub) 소스 보기</a>
  </td>
  <td>
    <a href="https://storage.googleapis.com/tensorflow_docs/docs-l10n/site/ko/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/download_logo_32px.png" />Download notebook</a>
  </td>
</table>

Note: 이 문서는 텐서플로 커뮤니티에서 번역했습니다. 커뮤니티 번역 활동의 특성상 정확한 번역과 최신 내용을 반영하기 위해 노력함에도
불구하고 [공식 영문 문서](https://www.tensorflow.org/?hl=en)의 내용과 일치하지 않을 수 있습니다.
이 번역에 개선할 부분이 있다면
[tensorflow/docs-l10n](https://github.com/tensorflow/docs-l10n/) 깃헙 저장소로 풀 리퀘스트를 보내주시기 바랍니다.
문서 번역이나 리뷰에 참여하려면
[docs-ko@tensorflow.org](https://groups.google.com/a/tensorflow.org/forum/#!forum/docs-ko)로
메일을 보내주시기 바랍니다.

이 문서는 [구글 코랩](https://colab.research.google.com/notebooks/welcome.ipynb)(Colaboratory) 노트북 파일입니다. 파이썬 프로그램을 브라우저에서 직접 실행할 수 있기 때문에 텐서플로를 배우고 사용하기 좋은 도구입니다:

1. 파이썬 런타임(runtime)에 연결하세요: 메뉴 막대의 오른쪽 상단에서 *CONNECT*를 선택하세요.
2. 노트북의 모든 코드 셀(cell)을 실행하세요: *Runtime* > *Run all*을 선택하세요.

더 많은 예제와 자세한 안내는 [텐서플로 튜토리얼](https://www.tensorflow.org/tutorials/)을 참고하세요.

먼저 프로그램에 텐서플로 라이브러리를 임포트합니다:

In [2]:
!pip install -q tensorflow-gpu==2.0.0-rc1
import tensorflow as tf

You should consider upgrading via the '/tmpfs/src/tf_docs_env/bin/python -m pip install --upgrade pip' command.


[MNIST 데이터셋](http://yann.lecun.com/exdb/mnist/)을 로드하여 준비합니다. 샘플 값을 정수에서 부동소수로 변환합니다:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

층을 차례대로 쌓아 `tf.keras.Sequential` 모델을 만듭니다. 훈련에 사용할 옵티마이저(optimizer)와 손실 함수를 선택합니다:

In [4]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

모델을 훈련하고 평가합니다:

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Train on 60000 samples
Epoch 1/5


   32/60000 [..............................] - ETA: 10:51 - loss: 2.3763 - accuracy: 0.0938

  768/60000 [..............................] - ETA: 30s - loss: 1.6923 - accuracy: 0.5104  

 1568/60000 [..............................] - ETA: 16s - loss: 1.3114 - accuracy: 0.6237

 2368/60000 [>.............................] - ETA: 12s - loss: 1.0789 - accuracy: 0.6913

 3168/60000 [>.............................] - ETA: 9s - loss: 0.9462 - accuracy: 0.7317 

 3968/60000 [>.............................] - ETA: 8s - loss: 0.8449 - accuracy: 0.7611

 4768/60000 [=>............................] - ETA: 7s - loss: 0.7777 - accuracy: 0.7806

 5600/60000 [=>............................] - ETA: 6s - loss: 0.7311 - accuracy: 0.7939

 6400/60000 [==>...........................] - ETA: 6s - loss: 0.6935 - accuracy: 0.8025

 7200/60000 [==>...........................] - ETA: 5s - loss: 0.6605 - accuracy: 0.8122

 8032/60000 [===>..........................] - ETA: 5s - loss: 0.6362 - accuracy: 0.8193

 8864/60000 [===>..........................] - ETA: 5s - loss: 0.6096 - accuracy: 0.8260

 9664/60000 [===>..........................] - ETA: 5s - loss: 0.5927 - accuracy: 0.8297

10464/60000 [====>.........................] - ETA: 4s - loss: 0.5751 - accuracy: 0.8355

11264/60000 [====>.........................] - ETA: 4s - loss: 0.5598 - accuracy: 0.8398

12096/60000 [=====>........................] - ETA: 4s - loss: 0.5456 - accuracy: 0.8433

12896/60000 [=====>........................] - ETA: 4s - loss: 0.5317 - accuracy: 0.8475

13728/60000 [=====>........................] - ETA: 4s - loss: 0.5159 - accuracy: 0.8515

14528/60000 [======>.......................] - ETA: 3s - loss: 0.5049 - accuracy: 0.8544

15360/60000 [======>.......................] - ETA: 3s - loss: 0.4955 - accuracy: 0.8568

16192/60000 [=======>......................] - ETA: 3s - loss: 0.4843 - accuracy: 0.8599

16992/60000 [=======>......................] - ETA: 3s - loss: 0.4758 - accuracy: 0.8631

17792/60000 [=======>......................] - ETA: 3s - loss: 0.4658 - accuracy: 0.8660

18592/60000 [========>.....................] - ETA: 3s - loss: 0.4589 - accuracy: 0.8681

19392/60000 [========>.....................] - ETA: 3s - loss: 0.4520 - accuracy: 0.8703

20224/60000 [=========>....................] - ETA: 3s - loss: 0.4430 - accuracy: 0.8725

21024/60000 [=========>....................] - ETA: 3s - loss: 0.4365 - accuracy: 0.8744

21856/60000 [=========>....................] - ETA: 3s - loss: 0.4313 - accuracy: 0.8757

22688/60000 [==========>...................] - ETA: 2s - loss: 0.4266 - accuracy: 0.8770

23520/60000 [==========>...................] - ETA: 2s - loss: 0.4204 - accuracy: 0.8789

24352/60000 [===========>..................] - ETA: 2s - loss: 0.4162 - accuracy: 0.8797

25184/60000 [===========>..................] - ETA: 2s - loss: 0.4104 - accuracy: 0.8813

26016/60000 [============>.................] - ETA: 2s - loss: 0.4060 - accuracy: 0.8826

26848/60000 [============>.................] - ETA: 2s - loss: 0.3998 - accuracy: 0.8845

27648/60000 [============>.................] - ETA: 2s - loss: 0.3953 - accuracy: 0.8857

28448/60000 [=============>................] - ETA: 2s - loss: 0.3913 - accuracy: 0.8872

29248/60000 [=============>................] - ETA: 2s - loss: 0.3866 - accuracy: 0.8885

30080/60000 [==============>...............] - ETA: 2s - loss: 0.3831 - accuracy: 0.8896

30912/60000 [==============>...............] - ETA: 2s - loss: 0.3803 - accuracy: 0.8905

31744/60000 [==============>...............] - ETA: 2s - loss: 0.3766 - accuracy: 0.8915

32576/60000 [===============>..............] - ETA: 2s - loss: 0.3729 - accuracy: 0.8927

33408/60000 [===============>..............] - ETA: 1s - loss: 0.3699 - accuracy: 0.8933

34240/60000 [================>.............] - ETA: 1s - loss: 0.3668 - accuracy: 0.8942

35072/60000 [================>.............] - ETA: 1s - loss: 0.3638 - accuracy: 0.8948

35872/60000 [================>.............] - ETA: 1s - loss: 0.3603 - accuracy: 0.8959

36704/60000 [=================>............] - ETA: 1s - loss: 0.3576 - accuracy: 0.8966

37536/60000 [=================>............] - ETA: 1s - loss: 0.3544 - accuracy: 0.8975

38368/60000 [==================>...........] - ETA: 1s - loss: 0.3512 - accuracy: 0.8983

39168/60000 [==================>...........] - ETA: 1s - loss: 0.3487 - accuracy: 0.8991

40000/60000 [===================>..........] - ETA: 1s - loss: 0.3456 - accuracy: 0.8998

40832/60000 [===================>..........] - ETA: 1s - loss: 0.3427 - accuracy: 0.9005

41632/60000 [===================>..........] - ETA: 1s - loss: 0.3394 - accuracy: 0.9014

42464/60000 [====================>.........] - ETA: 1s - loss: 0.3366 - accuracy: 0.9022

43296/60000 [====================>.........] - ETA: 1s - loss: 0.3342 - accuracy: 0.9028

44128/60000 [=====================>........] - ETA: 1s - loss: 0.3317 - accuracy: 0.9036

44928/60000 [=====================>........] - ETA: 1s - loss: 0.3295 - accuracy: 0.9044

45760/60000 [=====================>........] - ETA: 0s - loss: 0.3271 - accuracy: 0.9050

46560/60000 [======================>.......] - ETA: 0s - loss: 0.3247 - accuracy: 0.9055

47392/60000 [======================>.......] - ETA: 0s - loss: 0.3224 - accuracy: 0.9061

48192/60000 [=======================>......] - ETA: 0s - loss: 0.3203 - accuracy: 0.9068

49024/60000 [=======================>......] - ETA: 0s - loss: 0.3185 - accuracy: 0.9073

49824/60000 [=======================>......] - ETA: 0s - loss: 0.3163 - accuracy: 0.9078

50656/60000 [========================>.....] - ETA: 0s - loss: 0.3143 - accuracy: 0.9085

51456/60000 [========================>.....] - ETA: 0s - loss: 0.3123 - accuracy: 0.9091

52256/60000 [=========================>....] - ETA: 0s - loss: 0.3103 - accuracy: 0.9097

53088/60000 [=========================>....] - ETA: 0s - loss: 0.3082 - accuracy: 0.9102

53920/60000 [=========================>....] - ETA: 0s - loss: 0.3065 - accuracy: 0.9106

54720/60000 [==========================>...] - ETA: 0s - loss: 0.3049 - accuracy: 0.9110

55552/60000 [==========================>...] - ETA: 0s - loss: 0.3029 - accuracy: 0.9116

56352/60000 [===========================>..] - ETA: 0s - loss: 0.3011 - accuracy: 0.9121

57184/60000 [===========================>..] - ETA: 0s - loss: 0.2996 - accuracy: 0.9124

58016/60000 [============================>.] - ETA: 0s - loss: 0.2981 - accuracy: 0.9129

58816/60000 [============================>.] - ETA: 0s - loss: 0.2960 - accuracy: 0.9134

59616/60000 [============================>.] - ETA: 0s - loss: 0.2947 - accuracy: 0.9138

60000/60000 [==============================] - 4s 68us/sample - loss: 0.2941 - accuracy: 0.9140


Epoch 2/5
   32/60000 [..............................] - ETA: 5s - loss: 0.2524 - accuracy: 0.9062

  864/60000 [..............................] - ETA: 3s - loss: 0.1560 - accuracy: 0.9537

 1664/60000 [..............................] - ETA: 3s - loss: 0.1459 - accuracy: 0.9573

 2496/60000 [>.............................] - ETA: 3s - loss: 0.1456 - accuracy: 0.9551

 3328/60000 [>.............................] - ETA: 3s - loss: 0.1528 - accuracy: 0.9540

 4128/60000 [=>............................] - ETA: 3s - loss: 0.1502 - accuracy: 0.9552

 4960/60000 [=>............................] - ETA: 3s - loss: 0.1521 - accuracy: 0.9548

 5792/60000 [=>............................] - ETA: 3s - loss: 0.1546 - accuracy: 0.9539

 6624/60000 [==>...........................] - ETA: 3s - loss: 0.1577 - accuracy: 0.9535

 7424/60000 [==>...........................] - ETA: 3s - loss: 0.1563 - accuracy: 0.9539

 8224/60000 [===>..........................] - ETA: 3s - loss: 0.1576 - accuracy: 0.9539

 9024/60000 [===>..........................] - ETA: 3s - loss: 0.1575 - accuracy: 0.9539

 9856/60000 [===>..........................] - ETA: 3s - loss: 0.1543 - accuracy: 0.9545

10688/60000 [====>.........................] - ETA: 3s - loss: 0.1550 - accuracy: 0.9545

11488/60000 [====>.........................] - ETA: 3s - loss: 0.1550 - accuracy: 0.9546

12288/60000 [=====>........................] - ETA: 2s - loss: 0.1559 - accuracy: 0.9548

13120/60000 [=====>........................] - ETA: 2s - loss: 0.1545 - accuracy: 0.9547

13952/60000 [=====>........................] - ETA: 2s - loss: 0.1547 - accuracy: 0.9548

14784/60000 [======>.......................] - ETA: 2s - loss: 0.1532 - accuracy: 0.9554

15584/60000 [======>.......................] - ETA: 2s - loss: 0.1521 - accuracy: 0.9560

16384/60000 [=======>......................] - ETA: 2s - loss: 0.1531 - accuracy: 0.9559

17184/60000 [=======>......................] - ETA: 2s - loss: 0.1519 - accuracy: 0.9562

18016/60000 [========>.....................] - ETA: 2s - loss: 0.1523 - accuracy: 0.9560

18816/60000 [========>.....................] - ETA: 2s - loss: 0.1527 - accuracy: 0.9555

19616/60000 [========>.....................] - ETA: 2s - loss: 0.1528 - accuracy: 0.9554

20416/60000 [=========>....................] - ETA: 2s - loss: 0.1533 - accuracy: 0.9551

21248/60000 [=========>....................] - ETA: 2s - loss: 0.1522 - accuracy: 0.9552

22080/60000 [==========>...................] - ETA: 2s - loss: 0.1518 - accuracy: 0.9555

22912/60000 [==========>...................] - ETA: 2s - loss: 0.1520 - accuracy: 0.9556

23744/60000 [==========>...................] - ETA: 2s - loss: 0.1517 - accuracy: 0.9556

24544/60000 [===========>..................] - ETA: 2s - loss: 0.1503 - accuracy: 0.9562

25376/60000 [===========>..................] - ETA: 2s - loss: 0.1496 - accuracy: 0.9561

26176/60000 [============>.................] - ETA: 2s - loss: 0.1489 - accuracy: 0.9563

27008/60000 [============>.................] - ETA: 2s - loss: 0.1478 - accuracy: 0.9564

27840/60000 [============>.................] - ETA: 2s - loss: 0.1472 - accuracy: 0.9565

28640/60000 [=============>................] - ETA: 1s - loss: 0.1472 - accuracy: 0.9567

29472/60000 [=============>................] - ETA: 1s - loss: 0.1458 - accuracy: 0.9570

30272/60000 [==============>...............] - ETA: 1s - loss: 0.1461 - accuracy: 0.9569

31072/60000 [==============>...............] - ETA: 1s - loss: 0.1458 - accuracy: 0.9571

31904/60000 [==============>...............] - ETA: 1s - loss: 0.1448 - accuracy: 0.9575

32736/60000 [===============>..............] - ETA: 1s - loss: 0.1452 - accuracy: 0.9574

33568/60000 [===============>..............] - ETA: 1s - loss: 0.1451 - accuracy: 0.9573

34400/60000 [================>.............] - ETA: 1s - loss: 0.1451 - accuracy: 0.9573

35232/60000 [================>.............] - ETA: 1s - loss: 0.1446 - accuracy: 0.9574

36064/60000 [=================>............] - ETA: 1s - loss: 0.1443 - accuracy: 0.9575

36896/60000 [=================>............] - ETA: 1s - loss: 0.1434 - accuracy: 0.9578

37728/60000 [=================>............] - ETA: 1s - loss: 0.1426 - accuracy: 0.9580

38528/60000 [==================>...........] - ETA: 1s - loss: 0.1427 - accuracy: 0.9580

39328/60000 [==================>...........] - ETA: 1s - loss: 0.1423 - accuracy: 0.9582

40128/60000 [===================>..........] - ETA: 1s - loss: 0.1416 - accuracy: 0.9584

40960/60000 [===================>..........] - ETA: 1s - loss: 0.1411 - accuracy: 0.9585

41792/60000 [===================>..........] - ETA: 1s - loss: 0.1404 - accuracy: 0.9588

42624/60000 [====================>.........] - ETA: 1s - loss: 0.1413 - accuracy: 0.9584

43424/60000 [====================>.........] - ETA: 1s - loss: 0.1413 - accuracy: 0.9583

44224/60000 [=====================>........] - ETA: 0s - loss: 0.1413 - accuracy: 0.9584

45024/60000 [=====================>........] - ETA: 0s - loss: 0.1420 - accuracy: 0.9582

45824/60000 [=====================>........] - ETA: 0s - loss: 0.1423 - accuracy: 0.9582

46656/60000 [======================>.......] - ETA: 0s - loss: 0.1423 - accuracy: 0.9581

47456/60000 [======================>.......] - ETA: 0s - loss: 0.1422 - accuracy: 0.9582

48288/60000 [=======================>......] - ETA: 0s - loss: 0.1416 - accuracy: 0.9584

49120/60000 [=======================>......] - ETA: 0s - loss: 0.1410 - accuracy: 0.9586

49952/60000 [=======================>......] - ETA: 0s - loss: 0.1412 - accuracy: 0.9586

50784/60000 [========================>.....] - ETA: 0s - loss: 0.1409 - accuracy: 0.9585

51584/60000 [========================>.....] - ETA: 0s - loss: 0.1409 - accuracy: 0.9584

52416/60000 [=========================>....] - ETA: 0s - loss: 0.1409 - accuracy: 0.9584

53248/60000 [=========================>....] - ETA: 0s - loss: 0.1404 - accuracy: 0.9586

54048/60000 [==========================>...] - ETA: 0s - loss: 0.1407 - accuracy: 0.9584

54848/60000 [==========================>...] - ETA: 0s - loss: 0.1409 - accuracy: 0.9581

55680/60000 [==========================>...] - ETA: 0s - loss: 0.1409 - accuracy: 0.9582

56512/60000 [===========================>..] - ETA: 0s - loss: 0.1403 - accuracy: 0.9584

57344/60000 [===========================>..] - ETA: 0s - loss: 0.1402 - accuracy: 0.9585

58176/60000 [============================>.] - ETA: 0s - loss: 0.1397 - accuracy: 0.9586

58976/60000 [============================>.] - ETA: 0s - loss: 0.1397 - accuracy: 0.9586

59776/60000 [============================>.] - ETA: 0s - loss: 0.1397 - accuracy: 0.9586

60000/60000 [==============================] - 4s 62us/sample - loss: 0.1396 - accuracy: 0.9587


Epoch 3/5
   32/60000 [..............................] - ETA: 5s - loss: 0.0774 - accuracy: 0.9688

  864/60000 [..............................] - ETA: 3s - loss: 0.0939 - accuracy: 0.9699

 1696/60000 [..............................] - ETA: 3s - loss: 0.0961 - accuracy: 0.9711

 2528/60000 [>.............................] - ETA: 3s - loss: 0.0898 - accuracy: 0.9719

 3360/60000 [>.............................] - ETA: 3s - loss: 0.0844 - accuracy: 0.9729

 4160/60000 [=>............................] - ETA: 3s - loss: 0.0849 - accuracy: 0.9731

 4992/60000 [=>............................] - ETA: 3s - loss: 0.0913 - accuracy: 0.9720

 5792/60000 [=>............................] - ETA: 3s - loss: 0.0962 - accuracy: 0.9706

 6624/60000 [==>...........................] - ETA: 3s - loss: 0.1002 - accuracy: 0.9695

 7456/60000 [==>...........................] - ETA: 3s - loss: 0.1005 - accuracy: 0.9693

 8288/60000 [===>..........................] - ETA: 3s - loss: 0.0999 - accuracy: 0.9695

 9088/60000 [===>..........................] - ETA: 3s - loss: 0.1014 - accuracy: 0.9691

 9920/60000 [===>..........................] - ETA: 3s - loss: 0.1004 - accuracy: 0.9694

10752/60000 [====>.........................] - ETA: 3s - loss: 0.1001 - accuracy: 0.9696

11552/60000 [====>.........................] - ETA: 3s - loss: 0.1001 - accuracy: 0.9698

12352/60000 [=====>........................] - ETA: 2s - loss: 0.1006 - accuracy: 0.9696

13152/60000 [=====>........................] - ETA: 2s - loss: 0.1033 - accuracy: 0.9685

13984/60000 [=====>........................] - ETA: 2s - loss: 0.1038 - accuracy: 0.9683

14816/60000 [======>.......................] - ETA: 2s - loss: 0.1052 - accuracy: 0.9677

15648/60000 [======>.......................] - ETA: 2s - loss: 0.1058 - accuracy: 0.9675

16480/60000 [=======>......................] - ETA: 2s - loss: 0.1062 - accuracy: 0.9674

17312/60000 [=======>......................] - ETA: 2s - loss: 0.1069 - accuracy: 0.9674

18144/60000 [========>.....................] - ETA: 2s - loss: 0.1076 - accuracy: 0.9674

18976/60000 [========>.....................] - ETA: 2s - loss: 0.1078 - accuracy: 0.9671

19776/60000 [========>.....................] - ETA: 2s - loss: 0.1076 - accuracy: 0.9671

20576/60000 [=========>....................] - ETA: 2s - loss: 0.1061 - accuracy: 0.9676

21408/60000 [=========>....................] - ETA: 2s - loss: 0.1067 - accuracy: 0.9675

22240/60000 [==========>...................] - ETA: 2s - loss: 0.1065 - accuracy: 0.9674

23072/60000 [==========>...................] - ETA: 2s - loss: 0.1063 - accuracy: 0.9671

23872/60000 [==========>...................] - ETA: 2s - loss: 0.1057 - accuracy: 0.9673

24704/60000 [===========>..................] - ETA: 2s - loss: 0.1062 - accuracy: 0.9670

25536/60000 [===========>..................] - ETA: 2s - loss: 0.1059 - accuracy: 0.9670

26368/60000 [============>.................] - ETA: 2s - loss: 0.1055 - accuracy: 0.9673

27168/60000 [============>.................] - ETA: 2s - loss: 0.1056 - accuracy: 0.9676

28000/60000 [=============>................] - ETA: 1s - loss: 0.1056 - accuracy: 0.9676

28800/60000 [=============>................] - ETA: 1s - loss: 0.1054 - accuracy: 0.9676

29632/60000 [=============>................] - ETA: 1s - loss: 0.1059 - accuracy: 0.9673

30464/60000 [==============>...............] - ETA: 1s - loss: 0.1069 - accuracy: 0.9668

31296/60000 [==============>...............] - ETA: 1s - loss: 0.1072 - accuracy: 0.9668

32128/60000 [===============>..............] - ETA: 1s - loss: 0.1069 - accuracy: 0.9669

32960/60000 [===============>..............] - ETA: 1s - loss: 0.1067 - accuracy: 0.9670

33792/60000 [===============>..............] - ETA: 1s - loss: 0.1056 - accuracy: 0.9672

34624/60000 [================>.............] - ETA: 1s - loss: 0.1057 - accuracy: 0.9673

35456/60000 [================>.............] - ETA: 1s - loss: 0.1067 - accuracy: 0.9671

36288/60000 [=================>............] - ETA: 1s - loss: 0.1068 - accuracy: 0.9673

37120/60000 [=================>............] - ETA: 1s - loss: 0.1069 - accuracy: 0.9672

37952/60000 [=================>............] - ETA: 1s - loss: 0.1068 - accuracy: 0.9671

38784/60000 [==================>...........] - ETA: 1s - loss: 0.1068 - accuracy: 0.9672

39616/60000 [==================>...........] - ETA: 1s - loss: 0.1070 - accuracy: 0.9671

40416/60000 [===================>..........] - ETA: 1s - loss: 0.1063 - accuracy: 0.9672

41248/60000 [===================>..........] - ETA: 1s - loss: 0.1063 - accuracy: 0.9672

42080/60000 [====================>.........] - ETA: 1s - loss: 0.1063 - accuracy: 0.9672

42912/60000 [====================>.........] - ETA: 1s - loss: 0.1065 - accuracy: 0.9671

43744/60000 [====================>.........] - ETA: 1s - loss: 0.1064 - accuracy: 0.9671

44576/60000 [=====================>........] - ETA: 0s - loss: 0.1063 - accuracy: 0.9671

45408/60000 [=====================>........] - ETA: 0s - loss: 0.1062 - accuracy: 0.9671

46240/60000 [======================>.......] - ETA: 0s - loss: 0.1066 - accuracy: 0.9670

47072/60000 [======================>.......] - ETA: 0s - loss: 0.1066 - accuracy: 0.9670

47904/60000 [======================>.......] - ETA: 0s - loss: 0.1059 - accuracy: 0.9673

48736/60000 [=======================>......] - ETA: 0s - loss: 0.1053 - accuracy: 0.9675

49536/60000 [=======================>......] - ETA: 0s - loss: 0.1054 - accuracy: 0.9675

50368/60000 [========================>.....] - ETA: 0s - loss: 0.1052 - accuracy: 0.9676

51168/60000 [========================>.....] - ETA: 0s - loss: 0.1051 - accuracy: 0.9676

52000/60000 [=========================>....] - ETA: 0s - loss: 0.1050 - accuracy: 0.9677

52832/60000 [=========================>....] - ETA: 0s - loss: 0.1050 - accuracy: 0.9677

53632/60000 [=========================>....] - ETA: 0s - loss: 0.1053 - accuracy: 0.9676

54432/60000 [==========================>...] - ETA: 0s - loss: 0.1053 - accuracy: 0.9676

55264/60000 [==========================>...] - ETA: 0s - loss: 0.1053 - accuracy: 0.9677

56096/60000 [===========================>..] - ETA: 0s - loss: 0.1051 - accuracy: 0.9676

56928/60000 [===========================>..] - ETA: 0s - loss: 0.1051 - accuracy: 0.9677

57760/60000 [===========================>..] - ETA: 0s - loss: 0.1048 - accuracy: 0.9678

58592/60000 [============================>.] - ETA: 0s - loss: 0.1046 - accuracy: 0.9679

59392/60000 [============================>.] - ETA: 0s - loss: 0.1049 - accuracy: 0.9679

60000/60000 [==============================] - 4s 62us/sample - loss: 0.1046 - accuracy: 0.9680


Epoch 4/5
   32/60000 [..............................] - ETA: 5s - loss: 0.1120 - accuracy: 0.9375

  864/60000 [..............................] - ETA: 3s - loss: 0.0765 - accuracy: 0.9711

 1664/60000 [..............................] - ETA: 3s - loss: 0.0793 - accuracy: 0.9700

 2496/60000 [>.............................] - ETA: 3s - loss: 0.0847 - accuracy: 0.9700

 3328/60000 [>.............................] - ETA: 3s - loss: 0.0866 - accuracy: 0.9700

 4160/60000 [=>............................] - ETA: 3s - loss: 0.0830 - accuracy: 0.9726

 4992/60000 [=>............................] - ETA: 3s - loss: 0.0836 - accuracy: 0.9732

 5824/60000 [=>............................] - ETA: 3s - loss: 0.0863 - accuracy: 0.9727

 6656/60000 [==>...........................] - ETA: 3s - loss: 0.0870 - accuracy: 0.9728

 7488/60000 [==>...........................] - ETA: 3s - loss: 0.0858 - accuracy: 0.9736

 8320/60000 [===>..........................] - ETA: 3s - loss: 0.0832 - accuracy: 0.9742

 9120/60000 [===>..........................] - ETA: 3s - loss: 0.0814 - accuracy: 0.9742

 9952/60000 [===>..........................] - ETA: 3s - loss: 0.0813 - accuracy: 0.9741

10752/60000 [====>.........................] - ETA: 3s - loss: 0.0813 - accuracy: 0.9743

11584/60000 [====>.........................] - ETA: 3s - loss: 0.0827 - accuracy: 0.9743

12416/60000 [=====>........................] - ETA: 2s - loss: 0.0831 - accuracy: 0.9741

13248/60000 [=====>........................] - ETA: 2s - loss: 0.0813 - accuracy: 0.9749

14080/60000 [======>.......................] - ETA: 2s - loss: 0.0814 - accuracy: 0.9747

14912/60000 [======>.......................] - ETA: 2s - loss: 0.0816 - accuracy: 0.9747

15744/60000 [======>.......................] - ETA: 2s - loss: 0.0820 - accuracy: 0.9743

16544/60000 [=======>......................] - ETA: 2s - loss: 0.0803 - accuracy: 0.9749

17376/60000 [=======>......................] - ETA: 2s - loss: 0.0801 - accuracy: 0.9750

18176/60000 [========>.....................] - ETA: 2s - loss: 0.0810 - accuracy: 0.9746

19008/60000 [========>.....................] - ETA: 2s - loss: 0.0810 - accuracy: 0.9747

19840/60000 [========>.....................] - ETA: 2s - loss: 0.0812 - accuracy: 0.9744

20672/60000 [=========>....................] - ETA: 2s - loss: 0.0817 - accuracy: 0.9743

21504/60000 [=========>....................] - ETA: 2s - loss: 0.0815 - accuracy: 0.9747

22304/60000 [==========>...................] - ETA: 2s - loss: 0.0806 - accuracy: 0.9750

23136/60000 [==========>...................] - ETA: 2s - loss: 0.0810 - accuracy: 0.9749

23936/60000 [==========>...................] - ETA: 2s - loss: 0.0814 - accuracy: 0.9749

24736/60000 [===========>..................] - ETA: 2s - loss: 0.0813 - accuracy: 0.9750

25568/60000 [===========>..................] - ETA: 2s - loss: 0.0814 - accuracy: 0.9750

26368/60000 [============>.................] - ETA: 2s - loss: 0.0822 - accuracy: 0.9748

27200/60000 [============>.................] - ETA: 2s - loss: 0.0817 - accuracy: 0.9751

28032/60000 [=============>................] - ETA: 1s - loss: 0.0813 - accuracy: 0.9752

28864/60000 [=============>................] - ETA: 1s - loss: 0.0824 - accuracy: 0.9752

29664/60000 [=============>................] - ETA: 1s - loss: 0.0822 - accuracy: 0.9752

30464/60000 [==============>...............] - ETA: 1s - loss: 0.0826 - accuracy: 0.9751

31264/60000 [==============>...............] - ETA: 1s - loss: 0.0829 - accuracy: 0.9751

32064/60000 [===============>..............] - ETA: 1s - loss: 0.0832 - accuracy: 0.9752

32896/60000 [===============>..............] - ETA: 1s - loss: 0.0837 - accuracy: 0.9751

33696/60000 [===============>..............] - ETA: 1s - loss: 0.0843 - accuracy: 0.9750

34496/60000 [================>.............] - ETA: 1s - loss: 0.0842 - accuracy: 0.9750

35296/60000 [================>.............] - ETA: 1s - loss: 0.0842 - accuracy: 0.9748

36128/60000 [=================>............] - ETA: 1s - loss: 0.0847 - accuracy: 0.9747

36960/60000 [=================>............] - ETA: 1s - loss: 0.0844 - accuracy: 0.9748

37760/60000 [=================>............] - ETA: 1s - loss: 0.0844 - accuracy: 0.9748

38592/60000 [==================>...........] - ETA: 1s - loss: 0.0846 - accuracy: 0.9748

39392/60000 [==================>...........] - ETA: 1s - loss: 0.0845 - accuracy: 0.9747

40192/60000 [===================>..........] - ETA: 1s - loss: 0.0845 - accuracy: 0.9746

41024/60000 [===================>..........] - ETA: 1s - loss: 0.0846 - accuracy: 0.9746

41824/60000 [===================>..........] - ETA: 1s - loss: 0.0848 - accuracy: 0.9744

42624/60000 [====================>.........] - ETA: 1s - loss: 0.0847 - accuracy: 0.9744

43456/60000 [====================>.........] - ETA: 1s - loss: 0.0846 - accuracy: 0.9742

44288/60000 [=====================>........] - ETA: 0s - loss: 0.0841 - accuracy: 0.9745

45120/60000 [=====================>........] - ETA: 0s - loss: 0.0840 - accuracy: 0.9744

45952/60000 [=====================>........] - ETA: 0s - loss: 0.0836 - accuracy: 0.9745

46752/60000 [======================>.......] - ETA: 0s - loss: 0.0833 - accuracy: 0.9746

47584/60000 [======================>.......] - ETA: 0s - loss: 0.0835 - accuracy: 0.9745

48416/60000 [=======================>......] - ETA: 0s - loss: 0.0832 - accuracy: 0.9746

49216/60000 [=======================>......] - ETA: 0s - loss: 0.0832 - accuracy: 0.9746

50016/60000 [========================>.....] - ETA: 0s - loss: 0.0836 - accuracy: 0.9746

50816/60000 [========================>.....] - ETA: 0s - loss: 0.0836 - accuracy: 0.9747

51648/60000 [========================>.....] - ETA: 0s - loss: 0.0838 - accuracy: 0.9746

52480/60000 [=========================>....] - ETA: 0s - loss: 0.0838 - accuracy: 0.9745

53312/60000 [=========================>....] - ETA: 0s - loss: 0.0841 - accuracy: 0.9743

54144/60000 [==========================>...] - ETA: 0s - loss: 0.0840 - accuracy: 0.9744

54976/60000 [==========================>...] - ETA: 0s - loss: 0.0844 - accuracy: 0.9743

55808/60000 [==========================>...] - ETA: 0s - loss: 0.0843 - accuracy: 0.9743

56640/60000 [===========================>..] - ETA: 0s - loss: 0.0847 - accuracy: 0.9743

57472/60000 [===========================>..] - ETA: 0s - loss: 0.0851 - accuracy: 0.9742

58304/60000 [============================>.] - ETA: 0s - loss: 0.0854 - accuracy: 0.9742

59136/60000 [============================>.] - ETA: 0s - loss: 0.0860 - accuracy: 0.9741

59968/60000 [============================>.] - ETA: 0s - loss: 0.0859 - accuracy: 0.9741

60000/60000 [==============================] - 4s 62us/sample - loss: 0.0859 - accuracy: 0.9742


Epoch 5/5
   32/60000 [..............................] - ETA: 6s - loss: 0.0631 - accuracy: 0.9688

  864/60000 [..............................] - ETA: 3s - loss: 0.0663 - accuracy: 0.9780

 1696/60000 [..............................] - ETA: 3s - loss: 0.0613 - accuracy: 0.9794

 2528/60000 [>.............................] - ETA: 3s - loss: 0.0685 - accuracy: 0.9790

 3360/60000 [>.............................] - ETA: 3s - loss: 0.0658 - accuracy: 0.9801

 4160/60000 [=>............................] - ETA: 3s - loss: 0.0675 - accuracy: 0.9776

 4960/60000 [=>............................] - ETA: 3s - loss: 0.0702 - accuracy: 0.9772

 5760/60000 [=>............................] - ETA: 3s - loss: 0.0703 - accuracy: 0.9769

 6592/60000 [==>...........................] - ETA: 3s - loss: 0.0695 - accuracy: 0.9774

 7424/60000 [==>...........................] - ETA: 3s - loss: 0.0675 - accuracy: 0.9780

 8224/60000 [===>..........................] - ETA: 3s - loss: 0.0672 - accuracy: 0.9780

 9024/60000 [===>..........................] - ETA: 3s - loss: 0.0669 - accuracy: 0.9782

 9856/60000 [===>..........................] - ETA: 3s - loss: 0.0674 - accuracy: 0.9787

10656/60000 [====>.........................] - ETA: 3s - loss: 0.0675 - accuracy: 0.9783

11488/60000 [====>.........................] - ETA: 3s - loss: 0.0694 - accuracy: 0.9775

12320/60000 [=====>........................] - ETA: 2s - loss: 0.0706 - accuracy: 0.9769

13120/60000 [=====>........................] - ETA: 2s - loss: 0.0710 - accuracy: 0.9771

13952/60000 [=====>........................] - ETA: 2s - loss: 0.0714 - accuracy: 0.9772

14752/60000 [======>.......................] - ETA: 2s - loss: 0.0719 - accuracy: 0.9772

15584/60000 [======>.......................] - ETA: 2s - loss: 0.0715 - accuracy: 0.9772

16416/60000 [=======>......................] - ETA: 2s - loss: 0.0718 - accuracy: 0.9774

17216/60000 [=======>......................] - ETA: 2s - loss: 0.0720 - accuracy: 0.9776

18048/60000 [========>.....................] - ETA: 2s - loss: 0.0723 - accuracy: 0.9773

18880/60000 [========>.....................] - ETA: 2s - loss: 0.0719 - accuracy: 0.9775

19680/60000 [========>.....................] - ETA: 2s - loss: 0.0724 - accuracy: 0.9775

20480/60000 [=========>....................] - ETA: 2s - loss: 0.0724 - accuracy: 0.9775

21280/60000 [=========>....................] - ETA: 2s - loss: 0.0724 - accuracy: 0.9775

22112/60000 [==========>...................] - ETA: 2s - loss: 0.0725 - accuracy: 0.9776

22944/60000 [==========>...................] - ETA: 2s - loss: 0.0729 - accuracy: 0.9776

23776/60000 [==========>...................] - ETA: 2s - loss: 0.0726 - accuracy: 0.9778

24608/60000 [===========>..................] - ETA: 2s - loss: 0.0724 - accuracy: 0.9779

25440/60000 [===========>..................] - ETA: 2s - loss: 0.0728 - accuracy: 0.9776

26272/60000 [============>.................] - ETA: 2s - loss: 0.0727 - accuracy: 0.9776

27104/60000 [============>.................] - ETA: 2s - loss: 0.0730 - accuracy: 0.9773

27904/60000 [============>.................] - ETA: 2s - loss: 0.0730 - accuracy: 0.9773

28736/60000 [=============>................] - ETA: 1s - loss: 0.0726 - accuracy: 0.9774

29568/60000 [=============>................] - ETA: 1s - loss: 0.0721 - accuracy: 0.9776

30400/60000 [==============>...............] - ETA: 1s - loss: 0.0722 - accuracy: 0.9774

31232/60000 [==============>...............] - ETA: 1s - loss: 0.0716 - accuracy: 0.9775

32064/60000 [===============>..............] - ETA: 1s - loss: 0.0717 - accuracy: 0.9774

32864/60000 [===============>..............] - ETA: 1s - loss: 0.0718 - accuracy: 0.9774

33696/60000 [===============>..............] - ETA: 1s - loss: 0.0719 - accuracy: 0.9773

34496/60000 [================>.............] - ETA: 1s - loss: 0.0719 - accuracy: 0.9772

35328/60000 [================>.............] - ETA: 1s - loss: 0.0720 - accuracy: 0.9772

36160/60000 [=================>............] - ETA: 1s - loss: 0.0720 - accuracy: 0.9772

36992/60000 [=================>............] - ETA: 1s - loss: 0.0723 - accuracy: 0.9772

37824/60000 [=================>............] - ETA: 1s - loss: 0.0722 - accuracy: 0.9773

38656/60000 [==================>...........] - ETA: 1s - loss: 0.0725 - accuracy: 0.9773

39456/60000 [==================>...........] - ETA: 1s - loss: 0.0729 - accuracy: 0.9772

40288/60000 [===================>..........] - ETA: 1s - loss: 0.0729 - accuracy: 0.9773

41088/60000 [===================>..........] - ETA: 1s - loss: 0.0731 - accuracy: 0.9772

41920/60000 [===================>..........] - ETA: 1s - loss: 0.0731 - accuracy: 0.9772

42752/60000 [====================>.........] - ETA: 1s - loss: 0.0734 - accuracy: 0.9771

43584/60000 [====================>.........] - ETA: 1s - loss: 0.0731 - accuracy: 0.9772

44384/60000 [=====================>........] - ETA: 0s - loss: 0.0734 - accuracy: 0.9770

45184/60000 [=====================>........] - ETA: 0s - loss: 0.0734 - accuracy: 0.9770

46016/60000 [======================>.......] - ETA: 0s - loss: 0.0734 - accuracy: 0.9769

46848/60000 [======================>.......] - ETA: 0s - loss: 0.0734 - accuracy: 0.9770

47648/60000 [======================>.......] - ETA: 0s - loss: 0.0732 - accuracy: 0.9770

48480/60000 [=======================>......] - ETA: 0s - loss: 0.0732 - accuracy: 0.9770

49312/60000 [=======================>......] - ETA: 0s - loss: 0.0734 - accuracy: 0.9769

50144/60000 [========================>.....] - ETA: 0s - loss: 0.0733 - accuracy: 0.9769

50976/60000 [========================>.....] - ETA: 0s - loss: 0.0730 - accuracy: 0.9770

51808/60000 [========================>.....] - ETA: 0s - loss: 0.0731 - accuracy: 0.9770

52640/60000 [=========================>....] - ETA: 0s - loss: 0.0730 - accuracy: 0.9770

53472/60000 [=========================>....] - ETA: 0s - loss: 0.0729 - accuracy: 0.9771

54304/60000 [==========================>...] - ETA: 0s - loss: 0.0729 - accuracy: 0.9770

55136/60000 [==========================>...] - ETA: 0s - loss: 0.0729 - accuracy: 0.9769

55968/60000 [==========================>...] - ETA: 0s - loss: 0.0726 - accuracy: 0.9770

56768/60000 [===========================>..] - ETA: 0s - loss: 0.0727 - accuracy: 0.9770

57568/60000 [===========================>..] - ETA: 0s - loss: 0.0727 - accuracy: 0.9771

58400/60000 [============================>.] - ETA: 0s - loss: 0.0726 - accuracy: 0.9771

59200/60000 [============================>.] - ETA: 0s - loss: 0.0725 - accuracy: 0.9770

60000/60000 [==============================] - 4s 62us/sample - loss: 0.0724 - accuracy: 0.9771


10000/1 - 0s - loss: 0.0345 - accuracy: 0.9788


[0.06729823819857557, 0.9788]

훈련된 이미지 분류기는 이 데이터셋에서 약 98%의 정확도를 달성합니다. 더 자세한 내용은 [TensorFlow 튜토리얼](https://www.tensorflow.org/tutorials/)을 참고하세요.